validation-test/stdlib/Dictionary.swift

```diff
@@ -1312,6 +1312,30 @@ class ParallelArrayDictionary : NSDictionary {
   override var count: Int {
     return 4
   }
+
+  override func keyEnumerator() -> NSEnumerator {
+    return KeyEnumerator(dict: self)
+  }
+
+  class KeyEnumerator: NSEnumerator {
+    let dict: ParallelArrayDictionary
+    var idx: Int = 0
+
+    init(dict: ParallelArrayDictionary) {
+      self.dict = dict
+    }
+
+    override func nextObject() -> Any? {
+      idx += 1
+      switch idx {
+      case 1: return dict.keys[0].key0
+      case 2: return dict.keys[0].key1
+      case 3: return dict.keys[0].key2
+      case 4: return dict.keys[0].key3
+      default: return nil
+      }
+    }
+  }
 }
```

Comment:
Note to self: upon reflection this fix is dubious; it implies the motivation for this type is no longer happening. Need to investigate what the point of this *really* is.

Also note: the existing implementation appears to also have UB (strict aliasing violation).